packages/blocks/src/checkbox-input/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import { RawHTML, useEffect } from '@wordpress/element';
+import classnames from 'classnames';
+import { isEmpty } from 'lodash';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { useField } from '@crowdsignal/form';
+import { ErrorMessage, CheckboxAnswer, FormInputWrapper } from '../components';
+import { useColorStyles, useBorderStyles } from '@crowdsignal/styles';
+
+const CheckboxInput = ( { attributes, className } ) => {
+	const { error, fieldValue, onChange } = useField( {
+		fieldName: `q_${ attributes.clientId }[text]`,
+		validation: ( value ) => {
+			if ( attributes.mandatory && ! value ) {
+				return __( 'This field is required', 'blocks' );
+			}
+		},
+	} );
+
+	const isSelected = fieldValue === attributes.checkedText;
+	const value = isSelected
+		? attributes.checkedText
+		: attributes.uncheckedText;
+	const classes = classnames(
+		className,
+		'crowdsignal-forms-checkbox-input',
+		'is-inline',
+		{
+			'is-required': attributes.mandatory,
+			'is-error': error,
+		}
+	);
+
+	useEffect( () => {
```

Comment:
I think you'd want to pass `isSelected` as an `initialValue` prop to `useField` instead of doing this.

Reply:
That's also on me, sorry.
When I introduced the `initialValue` on the `useField` hook, I didn't realize that, at some point, we'd need to persist values from the store (navigating back and forward using the navigation bar).
The way the `initialValue` is implemented, it'll override the values from the store when navigating between the project's pages.
We need some refactoring there to use the `initialValue`.